### PR TITLE
add np to automate release steps

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?
- adds https://github.com/sindresorhus/np
- creates a `release` script, so that new releases can be managed by running `npm run release`

image

### CODE REVIEW

**Comment by JamieB-gu** [comment]
As far as I understand it, this is meant to be run from your repo directory on an up-to-date version of the `main` branch? Does that mean it requires pushing directly to `main`?

**Comment by sndrs** [comment]
you run it locally, so it will bump package.json and add a tag too, and push that. so you that's a good point, you'd need someone with write permission to `main` to run it

**Review by tjmw — APPROVED** [review]
This LGTM 👍 

(And thanks for the reminder that I've been meaning to give np a try on another project).

File: package.json [hunk 1]
```diff
@@ -44,5 +46,8 @@
       "dist"
     ],
     "preset": "ts-jest"
+  },
+  "np": {
+    "yarn": false
```
Comment (akash1810) [inline package.json:51]: Not sure if you want to be explicit about the branch too to enforce only `main` being used for publishing?

```suggestion
    "yarn": false,
    "branch": "main"
```

**Review by akash1810 — APPROVED** [review]
Nice! We've recently added `np` to the [cdk project](https://github.com/guardian/cdk/pull/23) and it's excellent. There's a slight caveat when you have branch protection enabled, as described [here](https://github.com/guardian/cdk/pull/25).  We've [not properly solved it](https://github.com/guardian/cdk/pulls?q=is%3Apr+is%3Aclosed+0.) yet; would be good to get your thoughts.

File: package.json (hunk 1 above)
Comment (sndrs) [reply package.json:51]: `main` or `master` are the defaults

**Comment by sndrs** [comment]
there no easy solution if your main branch is protected, because making sure `main` is in sync with the released version requires writing to `main` as part of the process. 

given this repo has no CD setup, what is the value of protecting `main`? (versus the benefit of the release process being able to write package metadata)

**Comment by JamieB-gu** [comment]
> I think you can remove `prepare` as the tarball will have the compiled code

Removing `prepare` would break the GitHub installs I think, because the tarball only applies to the registry? Unless there's a way to associate one with GitHub releases?

> given this repo has no CD setup, what is the value of protecting `main`?

Preventing developers from accidentally pushing to `main`? Unless there's another way to prevent that?

**Review by oliverlloyd — APPROVED** [review]
Shall we?

**Comment by oliverlloyd** [comment]
@sndrs any objections to me merging this one? np seems like a great lib and I'm keen to try it out!

**Pull request merged**